src/doc/layer_io.cpp

```diff
@@ -112,6 +114,10 @@ Layer* read_layer(std::istream& is, SubObjectsFromSprite* subObjects, const Seri
 {
   ObjectId id = read32(is);
   std::string name = read_string(is);
+  base::Uuid uuid;
+  if (serial >= SerialFormat::Ver3)
+    uuid = read_uuid(is);
```

Comment:
I'd try to save the UUID as the last new field of the layer stream (instead of shifting all old fields), i.e. saving the UUID after the user data.

Why this? Because probably and older decoder could read the layer data of a newer encoder, it means that a session of a future version can be recovered with previous version of Aseprite (losing the UUID, but at least reading all the other data).